### ISSUE TITLE
ESP32-C6 support status (IDFGH-9013)

### ISSUE THREAD

**Issue by igrr** [espressif/esp-idf#10423]
ESP-IDF v5.1, currently in development, will include initial support for ESP32-C6 chip.

For each feature, the status (Supported/Not supported) is given. An internal issue reference (such as "IDF-1234") is listed at the end of the feature description to help us keep this list up to date.

This list will be updated as the support for ESP32-C6 is being developed.

### Wi-Fi
- [x] 802.11bgn, HT40 – Supported
- [x] 802.11ax: 20MHz-only non-AP mode – Supported
- [x] Station, SoftAP, sniffer mode – Supported
- [x] UL/DL OFDMA (26/52/106/242 tone) – Supported
- [x] DL MU-MIMO，DL MU-MIMO within OFDMA – Supported
- [x] HE ER (HE Extended Range), DCM and ESP LR (ESP Long Range) – Supported
- [x] BSS Color – Supported
- [x] CSI (Channel State Information) – Supported
- [x] ESP-NOW – Supported
- [x] ESP-WiFi-Mesh – Supported
- [x] ESP-Touch v1/v2 – Supported
- [x] Wi-Fi Security – Supported
- [x] DPP (Device Provisioning Protocol) – Supported
- [x] Network-assisted roaming using 802.11k (Radio Measurements), 802.11v (BTM) specification and 802.11r (Fast BSS Transition) – Supported
- TWT
    - [x] Individual TWT – Supported
    - [ ] Broadcast TWT – Not supported (WIFI-5215)
- [x] Advanced DTIM Sleep – Supported
- [x] Power Save: modem sleep, light sleep – Supported
- [ ] FTM (Fine Time Measurement)  – Not supported  (WIFI-5144)
### Coexistence
- [x] Coexistence of Wi-Fi and Bluetooth LE - Supported
- [x] Coexistence of Wi-Fi and 802.15.4 - Supported
- [x] Coexistence of Bluetooth LE and 802.15.4 - Supported
- [x] External coexistence - Supported
### Bluetooth LE
- [x] Bluetooth 5 Controller (Bluetooth 5.3 Certified) - Supported
- [x] Bluedroid Host - Supported
- [x] NimBLE Host - Supported
- [x] ESP-BLE-MESH - Supported
- [x] BLE Modem Sleep and Light Sleep - Supported
- [x] Bluetooth Power Control - Supported
- [x] HCI - Supported
### 802.15.4
- [x] 802.15.4 Mac - Supported
- [x] 802.15.4 CSL - Supported
### Thread
- [x] Thread FTD/MTD - Supported
- [x] Thread RCP - Supported
- [x] Thread Border Router - Supported
### Zigbee
- [x] Zigbee 3.0 Home Automation - Supported
- [x] Zigbee RCP - Supported
- [x] Zigbee Gateway - Supported
### Peripherals
- ADC
    - [x] ADC Calibration - Supported
    - [x] ADC continuous mode driver - Supported
    - [x] ADC oneshot mode driver - Supported
- [x] Dedicated GPIO driver - Supported
- Ethernet driver
    - [x] SPI Ethernet driver - Supported
- [x] Event Task Matrix - Supported
- [x] GDMA - Supported
- [x] GPIO driver - Supported
    - [x] Glitch Filter - Supported
    - [x] RTC IO driver - Supported
- [x] GPIO sigma delta driver - Supported
- GPSPI
    - [x] SPI Master driver - Supported
    - [x] SPI Slave HD driver - Supported
    - [x] SPI Slave driver - Supported
- [x] GPTimer driver - Supported
- I2C
    - [x] I2C master driver - Supported
    - [x] I2C slave driver - Supported
- [x] I2S driver - Supported
    - I2S PDM rx mode
    - I2S PDM tx mode
    - I2S STD mode
    - I2S TDM mode
- [x] LCD driver - Supported
    - I2C LCD driver
    - SPI LCD driver
- [x] LEDC driver - Supported
    - [x] Gamma Curve Generation - Supported
- [x] MCPWM driver - Supported
- [x] PCNT driver - Supported
- [x] Parallel IO driver - Supported
- [x] RMT driver - Supported
- SD/SDIO/MMC driver
    - [x] SDSPI Host driver - Supported
- [x] SDIO Slave driver - Supported
- [x] SPI Flash driver - Supported
- Systimer driver
    - [x] Esptimer implementation over systimer - Supported
    - [x] OS tick port over systimer - Supported
- [x] TWAI driver - Supported
- [x] Temperature Sensor driver - Supported
- UART
    - [x] UART FIFO mode driver - Supported
    - [ ] UHCI (UART DMA mode) driver - Not supported (IDF-5337)
- [x] USB Serial JTAG - Supported
- [x] memory copy by DMA - Supported
### Power management
- [x] Auto Light Sleep - Supported
- [x] Deep Sleep - Supported
- [x] Dynamic Frequency Switch - Supported
- [x] Light Sleep - Supported
### Security Features
- [x] AES accelerator - Supported
- [x] Digital Signature driver - Supported
- [x] ECC accelerator - Supported
- [x] Flash encryption - Supported
- [x] HMAC driver - Supported
- [x] RNG - Supported
- [x] RSA hardware accelerator driver - Supported
- [x] SHA accelerator - Supported
- [x] Secure boot - Supported
- Trusted Execution Environment
    - [x] Access Permission Management - Supported
    - [x] TEE Controller - Supported
### System Features
- [x] Brownout Detector - Supported
- [x] Cache - Supported
- [x] Efuse controller driver - Supported
- [x] MMU - Supported
- MSPI
    - [ ] MSPI tuning (Flash & PSRAM DDR and/or over 80 MHz) - Not supported (IDF-5343)
    - [x] SPI Flash auto suspend - Not supported (IDF-5344)
- [ ] Timeout Module (New) - Not supported (IDF-5814)

_Note: Nested features can be considered as supported, if the parent feature is shown as "Supported", and it's not explicitly mentioned as "Not supported"._

**Comment by igrr** [comment]
Update 2023/01/02: added Wi-Fi features, ADC oneshot and continuous mode, RTC IO, SPI Master, SDSPI Host drivers.

**Comment by igrr** [comment]
Update 2023/01/13: added BLE features, Brownout Detector support

**Comment by igrr** [comment]
Update 2023/01/25: added HMAC, Digital Signature, SDIO slave, Bluetooth Power Control.

**Comment by mahavirj** [comment]
Update 2023/02/20: added RSA, Secure Boot, Flash Encryption, APM.

**Comment by gojimmypi** [comment]
I've confirmed wolfSSL encryption libraries work with the ESP-IDF v5.1 and the ESP32-C6. It is operational on my [fork](https://github.com/gojimmypi/wolfssl/tree/ED25519_SHA2_fix) and I plan to submit a PR soon to address https://github.com/wolfSSL/wolfssl/issues/6163 for full support.

**Comment by tom-borcin** [comment]
Update 2023/03/15: added Coexistence of Wi-Fi and Bluetooth LE, and Coexistence of Bluetooth LE and 802.15.4

**Comment by jack0c** [comment]
Update 2023/03/15: added Coexistence features and Power save features.

**Comment by rotilho** [comment]
Is it be possible to ZigBee / Matter, BLE and WiFi coexist?

**Comment by Hedda** [comment]
> Will it be possible ZigBee / Matter, BLE and WiFi to coexist?

Wondering this too about Zigbee and Thread? However as I understand it, Zigbee and Thread will intially be mutually exclusive, so can not currently use Zigbee and Thread simultaneously (at the same time)?

Not sure if feasable nativly on a single-core MCU but perhaps in future Zigbee and Thread could instead be timeshifted/timesliced to run concurrently using RCP (Radio Co-Processor) architecture design in where the Zigbee and Thread libraries/processes are off-loaded from the 802.15.4 SoC radio SoC to a host computer? 

https://openthread.io/platforms/co-processor

That is anyway how Silicon Labs and some others support using two different protocols at the same time on its 802.15.4 SoC radios.

PS: Off-topic but FYI; Home Assistant has experimental add-on for a such Zigbee and Therad implementation using off-loading running concurrently on their Silicon Labs based Home Assistant SkyConnect USB dongle with 802.15.4 SoC radio (and Home Assistent Yellow onboard radio based on the same 802.15.4 SoC radio SoC) in RCP mode.

**Comment by loraahd** [comment]
Hello,
Unfortunately I am not able to use my ESP32-C6 DEVKITC-1. That is because I am not able to install ESP-IDF V5.0.1 which is the only version that supports C6 targets. Please find attached the error message and the log file.

Anyone could give support please?
BR
 
[ADH Setup Log 2023-03-21 #001.txt](https://github.com/espressif/esp-idf/files/11026811/ADH.Setup.Log.2023-03-21.001.txt)
![Picture1](https://user-images.githubusercontent.com/128469243/226568590-dfa407eb-3458-47e9-8294-05594ffb0221.png)

**Comment by RuslanSpartak** [comment]
> Hello, Unfortunately I am not able to use my ESP32-C6 DEVKITC-1. That is because I am not able to install ESP-IDF V5.0.1 which is the only version that supports C6 targets. Please find attached the error message and the log file.
> 
> Anyone could give support please? BR
> 
> [ADH Setup Log 2023-03-21 #001.txt](https://github.com/espressif/esp-idf/files/11026811/ADH.Setup.Log.2023-03-21.001.txt) ![Picture1](https://user-images.githubusercontent.com/128469243/226568590-dfa407eb-3458-47e9-8294-05594ffb0221.png)

SDK 5.1 supports esp32c6, but it is not ready. You can use master brach for start of project to use esp32c6, but you will need to make some manuals manupulations to add for examples esp32c6 support in eclipse (as example).

**Comment by loraahd** [comment]
> > Hello, Unfortunately I am not able to use my ESP32-C6 DEVKITC-1. That is because I am not able to install ESP-IDF V5.0.1 which is the only version that supports C6 targets. Please find attached the error message and the log file.
> > Anyone could give support please? BR
> > [ADH Setup Log 2023-03-21 #001.txt](https://github.com/espressif/esp-idf/files/11026811/ADH.Setup.Log.2023-03-21.001.txt) ![Picture1](https://user-images.githubusercontent.com/128469243/226568590-dfa407eb-3458-47e9-8294-05594ffb0221.png)
> 
> SDK 5.1 supports esp32c6, but it is not ready. You can use master brach for start of project to use esp32c6, but you will need to make some manuals manipulations to add for examples esp32c6 support in eclipse (as example).

Thanks for you fast response @RuslanSpartak . would you kindly send me the link to the master branch with the needed manual manipulations for C6? do you have any idea on the date SDK 5.1 will begin supporting esp32c6, anytime soon?
BR
Achraf

**Comment by RuslanSpartak** [comment]
> > > Hello, Unfortunately I am not able to use my ESP32-C6 DEVKITC-1. That is because I am not able to install ESP-IDF V5.0.1 which is the only version that supports C6 targets. Please find attached the error message and the log file.
> > > Anyone could give support please? BR
> > > [ADH Setup Log 2023-03-21 #001.txt](https://github.com/espressif/esp-idf/files/11026811/ADH.Setup.Log.2023-03-21.001.txt) ![Picture1](https://user-images.githubusercontent.com/128469243/226568590-dfa407eb-3458-47e9-8294-05594ffb0221.png)
> > 
> > 
> > SDK 5.1 supports esp32c6, but it is not ready. You can use master brach for start of project to use esp32c6, but you will need to make some manuals manipulations to add for examples esp32c6 support in eclipse (as example).
> 
> Thanks for you fast response @RuslanSpartak . would you kindly send me the link to the master branch with the needed manual manipulations for C6? do you have any idea on the date SDK 5.1 will begin supporting esp32c6, anytime soon? BR Achraf

I'm also a customer that uses ESP32C6 module . You can get master branch from github. Which IDE you use, depends on that I can try to help you. About SDK5.1 I really don't know as I'm not emedded engineer of espressif.

**Comment by loraahd** [comment]
> 

I use IAR. I have no problem to use another one.
all I need is to flash my C6 and send commands to connect the C6 wifi to an access point. I don't really need to developpe any SW or FW.
thanks again for your kind support.

**Comment by RuslanSpartak** [comment]
> > 
> 
> I use IAR. I have no problem to use another one. all I need is to flash my C6 and send commands to connect the C6 wifi to an access point. I don't really need to developpe any SW or FW. thanks again for your kind support.

Are you able to install master branch instead of 5.0.1?

**Comment by tom-borcin** [comment]
Hello @rotilho, @Hedda, @loraahd and @RuslanSpartak,

this issue tracks only ESP32-C6 support status and for better visibility of update comments specifying what support was added, I had to hide your comments.

If you have any questions or need support, please feel free to open new issues [here](https://github.com/espressif/esp-idf/issues/new/choose) on GitHub or on [esp32.com](https://www.esp32.com) forum. Our developers and support team will be happy to help you.

Thank you for understanding, 

Tomas

**Comment by loraahd** [comment]
Don't know how to install it. Do you have a link to it?

**Comment by mrninhvn** [comment]
> Update 2023/03/15: added Coexistence of Wi-Fi and Bluetooth LE, and Coexistence of Bluetooth LE and 802.15.4

Hi, can Wi-Fi, BLE and 802.15.4 coexist?

**Comment by Espressif-liuuuu** [comment]
> > Update 2023/03/15: added Coexistence of Wi-Fi and Bluetooth LE, and Coexistence of Bluetooth LE and 802.15.4
> 
> Hi, can Wi-Fi, BLE and 802.15.4 coexist?

hi @mrninhvn , coexist of three modems together is not scheduled yet

**Comment by a-d-r-i-a-n-d** [comment]
Hi @Espressif-liuuuu does it mean that once the modems will coexist together we could have a Thread Border Router just with the ESP32-C6?

**Comment by igrr** [comment]
@a-d-r-i-a-n-d please see https://github.com/espressif/esp-thread-br/issues/9#issuecomment-1506402241 and leave a comment there if you have further questions regarding the Border Router solution.

**Comment by igrr** [comment]
Update 2023/04/18:  802.15.4 CSL, Thread Border Router, Parallel IO driver, LEDC gamma curve, systimer, cache and MMU are now supported.

**Comment by loraadh** [comment]
Hello,
I am not able to use my ESP32-C6 DEVKITC-1. That is because ESP-IDF V5.0.1 still does not supports C6 targets. Please find attached the error message.
Is there an upgrade that support C6 targets? any workaround?
![error2](https://github.com/espressif/esp-idf/assets/35763760/a2a8a9d1-9d74-4709-8fa2-173a2a9e9266)

BR,

**Comment by mickeyl** [comment]
@loraadh Espressif have stated that support for the C6 will debut with ESP-IDF 5.1. You can check out the `release/5.1` branch and it will already work pretty well with the C6.

**Comment by loraadh** [comment]
> @loraadh Espressif have stated that support for the C6 will debut with ESP-IDF 5.1. You can check out the `release/5.1` branch and it will already work pretty well with the C6.

thanks @mickeyl 
is the 5.1 version the same as the master version?
BR

**Comment by Hedda** [comment]
> is the 5.1 version the same as the master version?

No, 5.1 is the "release/v5.1" branch -> https://github.com/espressif/esp-idf/tree/release/v5.1

**Comment by loraadh** [comment]
> > is the 5.1 version the same as the master version?
> 
> No, 5.1 is the "release/v5.1" branch -> https://github.com/espressif/esp-idf/tree/release/v5.1
> 
> PS: Also follow [espressif/arduino-esp32#7852](https://github.com/espressif/arduino-esp32/issues/7852) and [espressif/arduino-esp32#7713](https://github.com/espressif/arduino-esp32/issues/7713) for updates on ESP-IDF 5.1 and ESP32-C6

thanks @Hedda,
in previous versions I used to do the installations via the online installer: https://dl.espressif.com/dl/esp-idf/?idf=4.4
Version ESP-IDF 5.1 is still not supported by that installer. Is there a user guide on how to install, flash/upload, and command the c6?
thanks 
BR

**Comment by Hedda** [comment]
For end-user questions and want support then you should post them to the esp32.com forum instead -> https://www.esp32.com

**Comment by Headcrabed** [comment]
Might be off-topic, but why c6's riscv trace support isn't on this table?

**Comment by igrr** [comment]
@Headcrabed indeed we did not include CPU Trace support into the plan, as we saw relatively few people using the similar feature on the previous Xtensa chips. It will get supported, eventually, especially if more users express interest.

**Comment by mickeyl** [comment]
Why was https://github.com/espressif/esp-idf/issues/10423#issuecomment-1549680635 marked as being offtopic?

**Comment by Renatomusica** [comment]
I Cant see the option to choose the ESP32-C6 Dev Kit in the very first VS code blink example. I am used to Program Texas Instruments And Microchip devices with C compilers and the approach is so different. It is my first time attempting with a ESP device that I just bought, so far this issue is really unfriendly already.. any help on this?
![Cant see ESP32-C6 on blink VS code Example](https://github.com/espressif/esp-idf/assets/22682422/ed4e483f-5f83-4d96-8efa-d8267b22788d)

**Comment by igrr** [comment]
@Renatomusica it is possible that the version of IDF you are using is one which doesn't contain C6 support. (For example, v5.0.x doesn't support C6). Please check that you have v5.1-rc1 installed, as that's the latest release candidate of v5.1. If you still run into an issue with VS Code after installing v5.1-rc1, please open an issue here: https://github.com/espressif/vscode-esp-idf-extension/issues

**Comment by Renatomusica** [comment]
> @Renatomusica it is possible that the version of IDF you are using is one which doesn't contain C6 support. (For example, v5.0.x doesn't support C6). Please check that you have v5.1-rc1 installed, as that's the latest release candidate of v5.1. If you still run into an issue with VS Code after installing v5.1-rc1, please open an issue here: https://github.com/espressif/vscode-esp-idf-extension/issues

Thank you I downloaded v5.1-rc1 and i can see device listed..
Now trying to build an example i run into all kinds of problems like asking for container location (i already set the folder where in this case the blink example should be set )  and also this
"Unable to determine what CMake generator to use. Please install or configure a preferred generator, or update settings.json, your Kit configuration or PATH variable." message that i tried to solve generating files from CMake extension but not working... do you have any source of guidance in order to accurately build an example w this system?

**Comment by emotasHil** [comment]
TWAI is marked as supported but actually only 1 of 2 TWAI controllers is supported.

**Comment by jack0c** [comment]
updated coexistence part, sorry since external coexistence is not supported in v5.1

**Comment by nunojpg** [comment]
With release-v5.1 or master I see 10-15ºC with the temperature sensor example, at a room at about 25ºC.

**Comment by AxelLin** [comment]
> updated coexistence part, sorry since external coexistence is not supported in v5.1

Is there any other progress update for C6?
Last update was 2 Months ago (https://github.com/espressif/esp-idf/issues/10423#issuecomment-1512679789),
I'm surprised the only update here is to mention that external coexistence is not supported.

@igrr

**Comment by Hedda** [comment]
@AxelLin Follow the status of ESP-IDF v5.1 as that upcoming will be first  ESP-IDF version to have initial support for the ESP32-C6:

https://github.com/espressif/arduino-esp32/issues/7713

https://github.com/espressif/arduino-esp32/issues/7852

Espressif ESP-IDF v5.1 RC 1 (ESP-IDF version 5.1 Release Candidate 1) was tagged in this GiHub repository just two weeks ago:

https://github.com/espressif/esp-idf/releases

https://github.com/espressif/esp-idf/releases/tag/v5.1-rc1

As both ESP32-C6 and ESP32-H2 are at the very top of the list of major features in that I think that you can safely assume that when the final ESP-IDF v5.1 release (i.e. "GA" release for "General Availability") is released as available then it will be officially supported by Espressif in that ESP-IDF v5.1 release. That is at least the industry standard for most software release life cycles:

https://en.wikipedia.org/wiki/Software_release_life_cycle

![image](https://github.com/espressif/esp-idf/assets/6320001/327f095d-a97f-4860-99b0-19cd5925c4d7)

**Comment by igrr** [comment]
ADC Calibration is now also supported on ESP32-C6.

The remaining features which aren't yet marked as supported will not be available in IDF v5.1 release. We will update the support plan for the remaining features before the next v5.1.1 release. In the meantime, if your project requires any of the features not listed as supported, please contact Espressif via business support channels.

**Comment by micbis** [comment]
Are there any plans to support also the second twai controller any time in the near future?

**Comment by AxelLin** [comment]
> With release-v5.1 or master I see 10-15ºC with the temperature sensor example, at a room at about 25ºC.

@nunojpg Please create a new issue to track this.

**Comment by marciogranzotto** [comment]
Missing from feature list: `OpenThread Sleepy End Device (SED)`

**Comment by jack0c** [comment]
updated:
- [x] External coexistence - Supported
- [x]  BLE Modem Sleep and Light Sleep - Supported

**Comment by embenix** [comment]
Sorry if I am repeating the same question as already asked by 2 other guys,
What are your plans for implementing the drivers for 2nd TWAI (`TWAI1`) port on ESP32C6? This could be a breakthrough in my opinion if 2nd TWAI port gets working sooner.

**Comment by wanckl** [comment]
@embenix sorry for confusing you, a new suit of TWAI driver is coming, 2nd TWAI will supported on new driver in new version of IDF at end 2023. sorry for long time waiting.

**Comment by kavehsgh** [comment]
Hi, Do you have any plan for the coexistence of Wifi-BLE-Zigbee?

**Comment by IsraelMek** [comment]
Hi,
Not really. Most likely we will be using the BLE. Since now we are at product development stage, we want to have the option of having more than one communication protocol, in case we want to go for another option. You  are very welcome if you have any suggestion.


Kind regards,
Israel Mekonnen

**Comment by HX2003** [comment]
Please refer to espressif/arduino-esp32#7713 regarding the BLE unified provisioning, so that the issue can be fixed. Thanks.

**Comment by AxelLin** [comment]
@igrr @jack0c 

Do you have update for the upcoming 5.1.2 release?
Do you have target timeline for feature complete? (Only 6 items are not yet supported)

**Comment by Massimiliano-solutiontech** [comment]
Hi,
I would like to know when the support for the `LP UART` will be available because I don't see the code in the `v5.2` but only on the `master` branch.

Thanks

**Comment by TRusselo** [comment]
Above it says I2S support is complete, but not working for me in current dev branch
![image](https://github.com/espressif/esp-idf/assets/1154815/833e023a-51d8-45f1-bd25-99eac4a49597)
![image](https://github.com/espressif/esp-idf/assets/1154815/06b82875-223c-4506-a57f-fa53cedcdb44)

**Comment by gojimmypi** [comment]
Happy to announce wolfSSL Hardware Acceleration Support for the ESP32-C6!  See [wolfSSL #6990](https://github.com/wolfSSL/wolfssl/pull/6990).  

I also have a [wolfSSL Managed Component](https://components.espressif.com/components/wolfssl/wolfssl) update in the works that will include this change once the PR is merged. The [wolfSSH](https://github.com/wolfSSL/wolfssh/tree/master/ide/Espressif) and [wolfMQTT](https://github.com/wolfSSL/wolfMQTT/tree/master) Espressif Managed components be added soon as well.

**Comment by jack0c** [comment]
> @igrr @jack0c
> 
> Do you have update for the upcoming 5.1.2 release? Do you have target timeline for feature complete? (Only 6 items are not yet supported)

The features supported in v5.1.2 were updated.
Here are some comments for the features not supported:
 Broadcast TWT – Not supported (WIFI-5215) - Still WIP, will support in IDF v5.3.
 FTM (Fine Time Measurement) – Not supported (WIFI-5144) - Still WIP, doing an ECO to support it, plan to support in IDF v5.3
 MSPI tuning (Flash & PSRAM DDR and/or over 80 MHz) - Not supported (IDF-5343) - Still WIP, not sure when to release
 SPI Flash auto suspend - Not supported (IDF-5344) - looks done, looks already in master 66bba5694e611980d2671b10ad0839ff1ad6e3e4
 Timeout Module (New) - Not supported (IDF-5814) - No plan yet

**Comment by Hedda** [comment]
Could you please add Zigbee NCP Host to the list? See -> https://github.com/espressif/esp-zigbee-sdk/issues/124

**Comment by txf-** [comment]
RTC Watchdog issues:

in menuconfig there is this option and description:
![image](https://github.com/espressif/esp-idf/assets/1192015/ee25302e-7be0-4a29-b971-199516784020)

As far as I can tell there `rtc_wdt_disable() `and `rtc_wdt_feed()`, do not apply to anything but the ESP32 and ESP32-S2 as those functions seem to be guarded by ifdefs for those devices.

Will this API ever be supported on other devices?

**Comment by tom-borcin** [comment]
Update 2024/01/17: 
- SPI Flash auto suspend - is supported
- Broadcast TWT - is in progress

Only remaining items are:
- FTM (Fine Time Measurement)
- UHCI (UART DMA mode) driver
- MSPI tuning (Flash & PSRAM DDR and/or over 80 MHz)
- Timeout Module (New)

These remaining items will be supported based on customer requests. We are closing the issue and unpinning it. Please open new issue if you have a feature request, question or any other problem.

**Comment by gojimmypi** [comment]
Woohoo! Congratulations to everyone that contributed to this massive update for ESP32C6!!

**Comment by AxelLin** [comment]
I'm wondering if all the supported features of C6 will be back port to 5.1 and 5.2 releases? (People mainly use release branches for real projects)

**Comment by Suxsem** [comment]
Hi, I'm using the relase/5.2 branch and I'm trying to use the ADC_ONESHOT with light sleep.

On boot, I initialize the ADC:
1) adc_oneshot_new_unit
2) adc_oneshot_config_channel
3) adc_cali_create_scheme_curve_fitting

Periodically, it wakes up from deep sleep and read the voltage:
4) adc_oneshot_read
5) adc_cali_raw_to_voltage

Unfortunately, only the first read is correct (in my case 1.8v); reads the occurs after light sleep are totally wrong (4v).

Is calibrated oneshoot adc supported after light sleep 1for esp32-c6?

**Comment by Suxsem** [comment]
@tom-borcin sorry for tagging you but can you please take a look at my previous comment? We have already produced some prototype pcbs and if adc is usable in light sleep we must find alternative solutions as soon as possible, thank you

**Comment by tom-borcin** [comment]
Hi @Suxsem, please open a new issue. We won't be able to track the progress of your issue here. Once you have an issue opened, we will triage it and assign engineer who will help you. Thank you!

**Comment by ilker-aktuna** [comment]
This post (issue) says Zigbee Home Automation is supported. Where can I find examples/documentation for that ?
I want to make a simple sensor using this. But I can't find any examples for joining a hub and communicating.

**Comment by igrr** [comment]
@ilker-aktuna Please check https://github.com/espressif/esp-zigbee-sdk/tree/5f70b2f/examples/esp_zigbee_HA_sample and https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/developing.html

**Comment by fillibar** [comment]
@igrr:
Those examples are for connecting between two of the ESP devices (one as end-device, one as coordinator) directly. They do not appear to show connecting to an existing ZigBee network/mesh that might be operated by a hub/controller (such as a Home Assistant or Hubitat). I think that is what @ilker-aktuna is looking for, ie: an end device that can communicate the data to that hub.

**Comment by igrr** [comment]
The readmes reference two ESP devices for simpler setup, but they can also be used with off-the-shelf hardware. For example, see https://github.com/espressif/esp-zigbee-sdk/issues/149. If you have further questions about ZigBee, please open an issue in esp-zigbee-sdk repository.

**Comment by ilker-aktuna** [comment]
> The readmes reference two ESP devices for simpler setup, but they can also be used with off-the-shelf hardware. For example, see [espressif/esp-zigbee-sdk#149](https://github.com/espressif/esp-zigbee-sdk/issues/149). 

@igrr 
I want to try these examples but I don't have development environment for these (VS)
Is it possible to have an Arduino version of these ?

In fact, I see that this Arduino example:
https://github.com/P-R-O-C-H-Y/arduino-esp32/blob/Zigbee-examples/libraries/ESP32/examples/Zigbee/Zigbee_Light_Bulb/Zigbee_Light_Bulb.ino

is the ported version of this:
https://github.com/espressif/esp-idf/blob/master/examples/zigbee/light_sample/HA_on_off_light/main/esp_zb_light.c

is that correct ?
if so, there is really not much difference.
Can we also use the example from here , to port to Arduino: 
https://github.com/espressif/esp-zigbee-sdk/tree/main/examples/esp_zigbee_HA_sample/HA_on_off_light

I just don't understand where the include files below will be found by Arduino IDE : 
#include "esp_zigbee_core.h"
#include "ha/esp_zigbee_ha_standard.h"

**Comment by diplfranzhoepfinger** [comment]
there is PSRAM on C6 ?

**Comment by slaesh** [comment]
> there is PSRAM on C6 ?

@diplfranzhoepfinger , sadly not: https://esp32.com/viewtopic.php?t=33703

**Comment by MichaelUray** [comment]
> UHCI (UART DMA mode) driver
Is there any timline for the UHCI implementation?

**Comment by tom-borcin** [comment]
@MichaelUray UHCI is implemented, we are testing it right now and it should be available in one month.
